wordpress-mobile/WordPress-iOS
# Zendesk Mobile: add mobile specific information to tickets

Fixes #9065 

**NOTES:**
- _Prerequisite_: You need a Zendesk account in order to view the tickets created via the app.
- Right now, the new Support view is only accessible if you are _logged in with a WP or social account_. And only via the navigation bar > Help. (Tip: I usually do Create Site > Help)
- There is some mobile information intentionally missing:
  - The `origin` tag. Requires `SupportSourceTag` support, which hasn't been done yet.
  - The `Site State` information. The exact values have not yet been determined.

**To test:**
- Go to the new Support view (as noted above).
- Submit a new ticket.
  - To view mobile tickets in Zendesk, use the filter noted in this [comment](https://wp.me/p9xubY-6U-p2/#comment-247). 
- Verify the following information on the ticket.

**Mobile ticket information:**

This list is in the form of Zendesk field names and the information the fields contain.

- `WP Mobile App Version Number`:
  - WordPress application version. Ex: `9.8`
- `WP Mobile App Blog Info`:
  - Information for all blogs, with each blog separated by "-------". See example below.
- `Free Space`:
  - Amount of free space available on the device. Ex: `1.72 GB`
- `WP Mobile App Other Info`:
  - Network Type, Carrier, Country Code, separated by \n. See example below.
- `System Status Report`:
  - Contains the most recent activity log (the same that is displayed in Support > Activity Logs > Current). See example below.
- `Tags`:
  - The `plan title` for each blog (free, personal, premium, business). To note: if there is no blog, there will be no plan tag.
  - `jetpack` or`wpcom`


**Blog information example:**
```
<Blog Name: bogusblog URL: https://bogusblog.wordpress.com XML-RPC: https://bogusblog.wordpress.com/xmlrpc.php wp.com account: bogusaccount blogId: 146666678 plan: Free (1)>
----------
<Blog Name: bogusblog2 URL: https://bogusblog2.wordpress.com XML-RPC: https://bogusblog2.wordpress.com/xmlrpc.php wp.com account: bogusaccount2 blogId: 169696924 plan: Free (1)>
```

**Network information example:**
```
Network Type: WiFi
Carrier: Verizon
Country Code: us
```

**Partial log example:**
```
2018/04/17 14:27:23:169  WPAnalytics session started
2018/04/17 14:27:23:177  ===========================================================================
2018/04/17 14:27:23:178  Launching WordPress for iOS 9.8 (9.8.0.2)...
2018/04/17 14:27:23:178  Crash count: 0
2018/04/17 14:27:23:178  Debug mode:  Debug
...
```

@aerych - thank you for volunteering to help with this one!
